tests/testutils/src/main/kotlin/io/element/android/tests/testutils/LongTask.kt

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright (c) 2023 New Vector Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.element.android.tests.testutils
+
+import app.cash.turbine.ReceiveTurbine
+import com.google.common.truth.Truth.assertThat
+import io.element.android.libraries.architecture.Async
+import kotlinx.coroutines.delay
+
+suspend fun <T> simulateLongTask(lambda: () -> Result<T>): Result<T> {
+    delay(1)
+    return lambda()
+}
+
+/**
+ * Await for an item, then ensure the item field, provided by the [asyncField], is a Loading.
+ * Then await for the next item, check for Success of Failure, depending on [shouldFail] and return it.
+ */
+suspend fun <State> ReceiveTurbine<State>.awaitLoadingAndAwait(
```

Comment:
I understand the urge to abstract this because there are many places in the codebase where such sequence of statements occurs.

Unfortunately the sole fact that there are many occurrences doesn't mean that creating a utility function is a good thing.

This change (in a somewhat similar way to `IsMe`) introduces complexity to the codebase (just look at how many files/lines this PR has grown by because of it).

Developers (and especially newcomers) won't easily be informed this utility function exist and might not end up using it. Reviewers will have to point it out during reviews increasing their load.

Newcomers reading our test code will wonder what `awaitLoadingAndAwait` is whilst they would have understood it right away the way it was before. This change will make the code obscure to their eyes.

I know the following example might seem a bit too "stretched" but how many times in a codebase statements like `something + somethingElse` occur?
So why don't we introduce a utility function  like:
```
fun <T> addSomethingToSomethingElse(something: T, somethingElse: T) {
    return something + somethingElse
}
```

Please not I have nothing against creating such kind of utility functions as private functions in a test file (should a developer chose to do so inside a test class).
My objection is making this kind of code part of foundational libraries over which our entire code will lay on.